test/dependency_source/test_poetry.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import annotations
+
+import subprocess
+import sys
+from pathlib import Path
+from textwrap import dedent
+from typing import Callable
+
+import pytest
+from packaging.version import Version
+
+from pip_audit._dependency_source import PoetrySource
+from pip_audit._fix import ResolvedFixVersion
+from pip_audit._service import ResolvedDependency
+
+
+@pytest.fixture
+def lock(tmp_path: Path) -> Callable:
+    def callback(*deps: str) -> Path:
+        metadata = """
+            [tool.poetry]
+            name = "poetry-demo"
+            version = "0.1.0"
+            description = ""
+            authors = ["someone <mail@example.com>"]
+
+            [tool.poetry.dependencies]
+            python = "^3.7"
+        """
+        metadata = dedent(metadata)
+        metadata += "\n".join(deps)
+        (tmp_path / "pyproject.toml").write_text(metadata)
+        cmd = [sys.executable, "-m", "poetry", "lock", "--no-update"]
+        subprocess.run(cmd, cwd=tmp_path).check_returncode()
+        lock_path = tmp_path / "poetry.lock"
+        assert lock_path.exists()
+        return lock_path
+
+    return callback
```

Comment:
Could you move this into `conftest.py`? That's where the rest of our test fixtures live 🙂

Reply:
Removed the fixture altogether ✅